src/librustc_mir/monomorphize/item.rs

```diff
@@ -65,7 +65,7 @@ pub trait MonoItemExt<'tcx>: fmt::Debug {
     fn instantiation_mode(&self, tcx: TyCtxt<'tcx>) -> InstantiationMode {
         let inline_in_all_cgus =
             tcx.sess.opts.debugging_opts.inline_in_all_cgus.unwrap_or_else(|| {
-                tcx.sess.opts.optimize != OptLevel::No
+                tcx.sess.opts.optimize != OptLevel::No && if tcx.sess.lto() == Lto::No
```

Comment:
There is an `if` after the `&&`. Why does this code compile?

Reply:
Oh good catch, and yeah I have no idea why this happened to compile. I'll push a fix later.

Reply:
Strange. I can't reproduce on play.rust-lang.org.